Update upstream distro to latest main and adapt integration (#383)
elastic/elastic-otel-php: docs/reference/edot-php/configuration.md

EDOT PHP also sets the `OTEL_PHP_AUTOLOAD_ENABLED` option to `true` by default. This turns on automatic instrumentation without requiring any changes to your application code.
Modifying this option will have no effect: EDOT will override it and enforce it as `true`.

> The distro package bundles multiple dependencies, such as the OpenTelemetry SDK, the auto-instrumentations, and their transitive dependencies. The monitored application might include dependencies that clash with the ones in the distro package, which could cause the application to malfunction. To prevent this, the distro uses **scoped** dependencies by default: a unique prefix is added to their namespaces. Because the PHP runtime supports runtime reflection, changing namespaces might theoretically be incompatible with some corner-case code. To fall back to the original (not scoped) dependencies, set `OTEL_PHP_SCOPED_DEPS_ENABLED` (php.ini `opentelemetry_distro.scoped_deps_enabled`) to `false`. See [Supportability](#supportability).

Reviewer: ```suggestion
EDOT PHP bundles multiple dependencies, such as the OpenTelemetry SDK, auto-instrumentations, and their transitive dependencies. This means that your application might include dependencies that clash with the bundled ones, which can cause it to malfunction. To prevent this, EDOT PHP uses scoped dependencies by default: a unique prefix is added to all bundled namespaces. Because PHP supports runtime reflection, this namespace change might be incompatible with some edge cases. To fall back to the original (unscoped) dependencies, set `OTEL_PHP_SCOPED_DEPS_ENABLED` (php.ini: `opentelemetry_distro.scoped_deps_enabled`) to `false`. Refer to [Supportability](#supportability) for more information.
```

Ugh, I hate it when there are too many possibilities 😄 I hope my suggestion explains clearly what might happen.

Also, I know what "distro" means, you know it, and the users also know but we don't use it in the docs for style reasons. It's "the EDOT PHP ✨ distribution ✨" or just "EDOT PHP". I know it's nitpicking! 🙂

_Deprecated aliases: `ELASTIC_OTEL_TRANSACTION_SPAN_ENABLED`, `ELASTIC_OTEL_TRANSACTION_SPAN_ENABLED_CLI`, `ELASTIC_OTEL_TRANSACTION_URL_GROUPS`_

### Attribute-based instrumentation configuration

{applies_to}`edot_php: ga 1.7.0`

| Option(s)                   | Default | Accepted values   | Description                                                                                                                                                |
| --------------------------- | ------- | ----------------- | -------------------------------------------------------------------------------------------------------------------------------------------------------- |

> | OTEL_PHP_ATTR_HOOKS_ENABLED | `false`   | `true` or `false`     | Enables `#[WithSpan]` / `#[SpanAttribute]` attribute-based span creation. See [Attribute-based instrumentation](attribute-instrumentation.md). |

Reviewer: ```suggestion
| OTEL_PHP_ATTR_HOOKS_ENABLED | `false`   | `true` or `false`     | Turns on `#[WithSpan]` / `#[SpanAttribute]` attribute-based span creation. Refers to [Attribute-based instrumentation](attribute-instrumentation.md). |
```

as long as EDOT PHP is on version 1.2.0 or later, even if `sampling_rate` applies to 9.3 and later.
:::

### Supportability

{applies_to}`edot_php: ga 1.7.0`

| Option(s)                    | Default | Accepted values   | Description                                                                                                                            |
| ---------------------------- | ------- | ----------------- | ------------------------------------------------------------------------------------------------------------------------------------ |

> | OTEL_PHP_SCOPED_DEPS_ENABLED | `true`    | `true` or `false`     | Controls whether the distro uses scoped or original (not scoped) dependencies. See [Special considerations](#special-considerations). |

Reviewer: ```suggestion
| OTEL_PHP_SCOPED_DEPS_ENABLED | `true`    | `true` or `false`     | Controls whether EDOT PHP uses scoped or original (not scoped) dependencies. Refer to [Special considerations](#special-considerations). |
```
✨

_Deprecated alias: `ELASTIC_OTEL_SCOPED_DEPS_ENABLED`_

## File-based configuration (declarative)

{applies_to}`edot_php: ga 1.7.0`

As an alternative to environment variables, you can configure the SDK using a YAML configuration file by setting the `OTEL_CONFIG_FILE` environment variable:

```bash
export OTEL_CONFIG_FILE=/path/to/otel-config.yaml
```

When `OTEL_CONFIG_FILE` is set:

- The SDK reads all configuration from the YAML file instead of individual `OTEL_*` environment variables.
- Environment variable substitution (`${MY_VAR:-default}`) is supported within the YAML file.
- Central configuration (OpAMP) is automatically disabled — file-based and remote configuration are mutually exclusive.
- Distro-specific options (`OTEL_PHP_*`) continue to work as they are native extension options, independent of the SDK.

### Distro resource detector

> The distro provides a `distro` resource detector that adds `telemetry.distro.name` and `telemetry.distro.version` resource attributes. To activate it in file-based configuration, add it to the `resource.detection/development.detectors` section:

Reviewer: Please remove all "distros". See my earlier comment for alternatives

```yaml
file_format: "1.0-rc.2"

resource:
  attributes:
    - name: service.name
      value: my-service
  detection/development:
    detectors:
      - distro: {}

propagator:
  composite:
    - tracecontext:
    - baggage:

tracer_provider:
  processors:
    - batch:
        exporter:
          otlp_http:
            endpoint: http://localhost:4318/v1/traces

meter_provider:
  readers:
    - periodic:
        exporter:
          otlp_http:
            endpoint: http://localhost:4318/v1/metrics

logger_provider:
  processors:
    - batch:
        exporter:
          otlp_http:
            endpoint: http://localhost:4318/v1/logs
```

> For the full YAML schema, see the [OpenTelemetry Configuration Schema](https://github.com/open-telemetry/opentelemetry-configuration/blob/main/schema-docs.md).

Reviewer: ```suggestion
For the full YAML schema, refer to the [OpenTelemetry Configuration Schema](https://github.com/open-telemetry/opentelemetry-configuration/blob/main/schema-docs.md).
```

### Limitations

- Central configuration (OpAMP) is not available when file-based configuration is active.

> - Resource detectors registered via `Registry::registerResourceDetector()` (for example, cloud provider detectors from `opentelemetry-php-contrib`) are not automatically active. They must provide a `ComponentProvider` and be explicitly listed in the YAML `resource.detection/development.detectors` section.

Reviewer: ```suggestion
- Resource detectors registered using `Registry::registerResourceDetector()` (for example, cloud provider detectors from `opentelemetry-php-contrib`) are not automatically active. They must provide a `ComponentProvider` and be explicitly listed in the YAML `resource.detection/development.detectors` section.
```
